src/gtc/passes/oir_pipeline.py

```diff
@@ -50,18 +51,32 @@ def hash_step(step: Callable) -> int:
     return hash(step)
 
 
-class OirPipeline:
+class OirPipeline(ABC):
+    @abstractmethod
+    def __hash__(self) -> int:
+        pass
+
+    @abstractmethod
+    def __repr__(self) -> str:
+        pass
+
+    @abstractmethod
+    def run(self, oir: oir.Stencil) -> oir.Stencil:
+        pass
+
+
+class DefaultPipeline(OirPipeline):
     """
     OIR passes pipeline runs passes in order and allows skipping.
 
     May only call existing passes and may not contain any pass logic itself.
     """
 
-    def __init__(self, node: oir.Stencil):
-        self.oir = node
-        self._cache: Dict[Tuple[int, ...], oir.Stencil] = {}
+    def __init__(self, *, skip: Optional[Sequence[PASS_T]] = None):
```

Comment:
This is not strictly part of this PR, but `PASS_T` does not follow the naming conventions for a typing hint representing a class. It should be just `Pass` or `AnyPass` or  `PassT`  if it were defined as a `TypeVar`.